File: pset6/caesar/caesar.py
from sys import argv
from cs50 import get_string

if (len(argv) == 2):
    # Prompt user for inputs
    k = int(argv[1])
    p = get_string("plaintext: ")
    # Caesar's cipher
    print("ciphertext: ", end="")
    for c in p:
        if c.isalpha():
            if c.isupper():
                print(f"{chr((ord(c) - 65 + k) % 26 + 65)}", end="")
            else:
                print(f"{chr((ord(c) - 97 + k) % 26 + 97)}", end="")
        else:
            print(c, end="")
    print()
# Prints an error message
else:
    print("Usage: python caesar.py k")
    exit(1)

File: pset2/caesar/caesar.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, string argv[])
{
    int k;
    string p;
    if (argc == 2)
    {
        //Prompt user for inputs
        k = atoi(argv[1]);
        p = get_string("plaintext: ");
        //Caesar's cipher
        printf("ciphertext: ");
        for (int i = 0, n = strlen(p); i < n; i++)
        {
            //Check for alphabtical
            if (isalpha(p[i]))
            {
                if (isupper(p[i]))
                {
                    p[i] -= 65;
                    p[i] = (p[i] + k) % 26;
                    p[i] += 65;
                    printf("%c", p[i]);
                }
                else
                {
                    p[i] -= 97;
                    p[i] = (p[i] + k) % 26;
                    p[i] += 97;
                    printf("%c", p[i]);
                }
            }
            else
            {
                printf("%c", p[i]);
            }
        }
        printf("\n");
    }
    //Print an error message
    else
    {
        return 1;
    }
}